src/server/authMiddleware.ts

```diff
@@ -72,19 +95,32 @@ const createAuthMiddleware =
 	async (req: NextRequest) => {
 		console.debug('Auth middleware starts');
 
+		const { publicRoutes, privateRoutes, ...restOptions } = options;
+		if (publicRoutes && privateRoutes) {
+			console.warn(
+				'Both publicRoutes and privateRoutes are defined. Ignoring privateRoutes.'
+			);
+		}
+
+		const effectiveOptions = {
```

Comment:
just an small code style opinion,
 
 I think that a new "effective" object is not readable and future to error prone for few reasons:
  -future developer may use the `options` 
  -  the name `effective` doesn't really communicate this message of "use this new object"
  - what will happen if we have more cases like this where we need to add it?

I think a better approach will be to pass public routes and private routes to the `isPublicRoute` function

thinking about it more, this log line and all the logic can (and should) be done inside `isPublicRoute`, this should not be a concern of this function